react/features/keyboard-shortcuts/components/KeyboardShortcutsDialog.web.js

```diff
@@ -0,0 +1,91 @@
+import Lozenge from '@atlaskit/lozenge';
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+
+import { Dialog } from '../../base/dialog';
+import { translate } from '../../base/i18n';
+
+/**
+ * Implements a React {@link Component} which displays a dialog describing
+ * registered keyboard shortcuts.
+ *
+ * @extends Component
+ */
+class KeyboardShortcutsDialog extends Component {
+    /**
+     * {@code KeyboardShortcutsDialog} component's property types.
+     *
+     * @static
+     */
+    static propTypes = {
+        /**
+         * A Map with keyboard keys as keys and translation keys as values.
+         */
+        shortcutDescriptions: PropTypes.object,
+
+        /**
+         * Invoked to obtain translated strings.
+         */
+        t: PropTypes.func
+    };
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        const { shortcutDescriptions } = this.props;
+        const shortcuts = [];
+
+        for (const [ keyboardKey, translationKey ] of shortcutDescriptions) {
+            shortcuts.push(
+                this._renderShortcutsListItem(keyboardKey, translationKey));
+        }
+
+        return (
+            <Dialog
+                cancelTitleKey = { 'dialog.close' }
+                submitDisabled = { true }
+                titleKey = 'keyboardShortcuts.keyboardShortcuts'
+                width = 'small'>
+                <div
+                    id = 'keyboard-shortcuts'>
+                    <ul
+                        className = 'shortcuts-list'
+                        id = 'keyboard-shortcuts-list'>
+                        { shortcuts }
```

Comment:
Very minor question. Wouldn't the es6 way be to use Object.keys(shortcutDescriptions).map((keyboardKey, translationKey) => this._render...) or something in this direction instead of an additional array?

Reply:
Whenever I've used Object.keys before I was told to use for..of instead. To be honest, I think either works and it comes down to style preference; I feel they're not far from each other in readability and I would assume performance differences are minor enough for the context. Because the object is actually a map it would also be possible to do map.keys() or Array.from() or other methods. If you prefer a method chaining approach, which I actually do, I can make the change--it's almost no effort to do so.

Reply:
Nevermind, I decided to just change it while I was looking at the package-lock stuff. Just let me know if you prefer a different way (or a different implementation entirely).